### PR TITLE
feat: zaps

### PR DESCRIPTION
**Pull request by verbiricha**
- [ ] Embed zapped notes

### CODE REVIEW

File: src/Hash.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as secp from "@noble/secp256k1";
+import { sha256 as hash } from '@noble/hashes/sha256';
+
+export const sha256 = (str: string) => {
```
Comment (v0l) [inline src/Hash.tsx:4]: You can put this in Util?

**Review by v0l — APPROVED** [review]
Awesome!

**Comment by verbiricha** [comment]
Follow-up PR implements zap sending (WIP) and has some improvements on this https://github.com/v0l/snort/pull/80

File: src/Element/Zap.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  if (zapRequest) {
+    const rawEvent: TaggedRawEvent = JSON.parse(zapRequest);
+    if (Array.isArray(rawEvent)) {
+      // old format, ignored
+      const rawDescriptionTag = rawEvent.find(a => a[0] === 'application/nostr')
+      const rawDescription = rawDescriptionTag && rawDescriptionTag[1]
+      const request = typeof rawDescription === 'string' ? JSON.parse(rawDescription) : rawDescription
+      return request?.pubkey
+    }
+    //const metaHash = sha256(zapRequest);
```
Comment (v0l) [inline src/Element/Zap.tsx:44]: This is wrong, were not using this format anymore, can you revert to previous?

The `description` tag contains the zap note and the hash of this should match the description hash of the invoice

**Review by v0l — CHANGES_REQUESTED** [review]
We should also validate that the `ZapReceipt` pubkey matches the pubkey listed in the recipients lightning service, otherwise i can just fake zaps to people

**Review by v0l — APPROVED** [review]

**Pull request merged**